RPIPs/RPIP-59.md

```diff
@@ -1,90 +0,0 @@
-
-Defines the queue structures that govern validator deposits and the overall deposit process within Rocket Pool. Both a standard and express queue structure are present. Access to the express queue is managed via tickets. A limited number of tickets are provided to all new nodes, and an amount of tickets are provided to existing nodes at the time of the upgrade based on their amount of bonded ETH. The express queue is designed to progress faster than the standard queue.
-
-This proposal also changes the deposit mechanics: In case of a queue, the initial stake transaction happens only once ETH is assigned. This makes it possible to exit from the queue and receive ETH credit up until the validator is dequeued.
-
-## Specification
-
-### Deposit queue specification
-ETH from the deposit pool SHALL be matched with validator deposits from queues as follows:
-- There SHALL be a `standard_queue`
-  - When adding a validator, users MAY place their deposit on the `standard_queue`  
-- There SHALL be an `express_queue`
-  - When adding a validator, users MAY place their deposit on the `express_queue` by spending one `express_queue_ticket`
-- When matching ETH from the deposit pool to queued deposits:
-  - First, ETH SHALL be matched to the oldest deposit in the `express_queue`; this is repeated `express_queue_rate` times
-  - Next, ETH SHALL be matched to the oldest deposit in the `standard_queue`
-  - This sequence SHALL be repeated indefinitely until both queues are empty
-    - If one queue is empty, those matches SHALL be skipped
-- Each node SHALL be provided `express_queue_tickets` equal to `express_queue_tickets_base_provision` (this includes newly created nodes)
-- Each node SHALL be provided additional `express_queue_tickets` equal to `(bonded ETH in legacy minipools)/4` (this will always be zero for newly created nodes)
-- The initial settings SHALL be:
-  - `express_queue_rate`: 2
-  - `express_queue_tickets_base_provision`: 2
-
-### Deposit mechanics specification
-- A node operator MUST take 2 actions to start a validator: `deposit` and `stake`
-
-#### `deposit` Transaction
-- `deposit` SHALL place the Node Operator ETH in the deposit pool (where it can be used in validators as needed) and place the validator in a queue as described [above](#deposit-queue-specification)
-- The following values for the validator SHALL be stored on chain:
-    - the public key of the validator
-    - the BLS signature over the public key, the withdrawal credentials (the megapool address of the node operator), and the amount (1 ETH) as required by the deposit contract
-    - the deposit message root as required by the deposit contract
-- The transaction SHALL validate the provided values and revert if they do not match
-- `deposit` SHALL assign deposits as described below 
-
-#### Assigning ETH from the Deposit Pool
-- As ETH enters the deposit pool, it SHALL be assigned to valdiators from the queue by sending 32 ETH to the associated megapool contract
```

Comment:
Might be this is obvious, and I just don't have enough understanding of the protocol. Why is this 32 ETH? Shouldn't it be 32 - bond provided by the operator?

Reply:
The previous section placed the bond in the deposit pool (for use by the front of the queue), so the deposit pool is always sending 32.

Reply:
How does this work if there are no rETH depositors, and several NOs trying to start validators? Do they all have credit, but if there are enough to hit 32 ETH, one can spin up a new validator? That would seem to break the whole 'you can exit the queue at any point and withdraw' thing defined elsewhere?

Reply:
That's how it works, and you exit for credit, not ETH. There is a provision to trade credit for rETH (again, not ETH directly)

Reply:
Ah, okay. So they can exit the queue early, but they end up with rETH rather than ETH?